benchmarks/bench_plot_onnxruntime_logreg.py

```diff
@@ -0,0 +1,230 @@
+                           figsize=(ncols * 4, nrows * 4))
+    pos = 0
+    row = 0
+    for n_obs in sorted(set(df.n_obs)):
+        for fit_intercept in sorted(set(df.fit_intercept)):
+            pos = 0
+            for method in sorted(set(df.method)):
+                a = ax[row, pos]
+                if row == ax.shape[0] - 1:
+                    a.set_xlabel("N features", fontsize='x-small')
+                if pos == 0:
+                    a.set_ylabel("Time (s) n_obs={}\nfit_intercept={}".format(n_obs, fit_intercept),
+                                 fontsize='x-small')
+
+                color = 'b'
+                subset = df[(df.method == method) & (df.n_obs == n_obs) &
+                            (df.fit_intercept == fit_intercept)]
+                if subset.shape[0] == 0:
+                    continue
+                subset = subset.sort_values("nfeat")
+                if verbose:
+                    print(subset)
+                label = "skl"
+                subset.plot(x="nfeat", y="time_skl", label=label, ax=a,
+                            logx=True, logy=True, c=color, style='--')
+                label = "ort"
+                subset.plot(x="nfeat", y="time_ort", label=label, ax=a,
+                            logx=True, logy=True, c=color)
+
+                a.legend(loc=0, fontsize='x-small')
+                if row == 0:
+                    a.set_title("method={}".format(method), fontsize='x-small')
+                pos += 1
+            row += 1
+
+    plt.suptitle(
+        "Benchmark for LogisticRegression sklearn/onnxruntime", fontsize=16)
+
+
+def run_bench(repeat=100, verbose=False):
+    n_obs = [1, 10]
+    methods = ['predict', 'predict_proba']
+    n_features = [1, 5, 10, 20, 50, 100, 200]
+    fit_intercepts = [True, False]
+
+    start = time()
+    results = bench(n_obs, n_features, fit_intercepts, methods,
+                    repeat=repeat, verbose=verbose)
+    end = time()
+
+    results_df = pandas.DataFrame(results)
+    print("Total time = %0.3f sec\n" % (end - start))
+
+    # plot the results
+    plot_results(results_df, verbose=verbose)
+    return results_df
+
+
+if __name__ == '__main__':
+    from datetime import datetime
+    import sklearn
+    import numpy
+    import onnx
+    import onnxruntime
+    import skl2onnx
+    df = pandas.DataFrame([
+        {"name": "date", "version": str(datetime.now())},
+        {"name": "numpy", "version": numpy.__version__},
+        {"name": "scikit-learn", "version": sklearn.__version__},
+        {"name": "onnx", "version": onnx.__version__},
+        {"name": "onnxruntime", "version": onnxruntime.__version__},
+        {"name": "skl2onnx", "version": skl2onnx.__version__},
+    ])
+    df.to_csv("bench_plot_onnxruntime_logreg.time.csv", index=False)
+    print(df)
+    df = run_bench(verbose=True)
+    plt.savefig("bench_plot_onnxruntime_logreg.png")
+    df.to_csv("bench_plot_onnxruntime_logreg.csv", index=False)
+    # plt.show()
```

Comment:
Why is this commented code left here?

Reply:
up to the user, I don't think that makes a big difference.

Reply:
done